sdk/core/System.ClientModel/tests/client/ModelReaderWriter/Models/DiscriminatorSet/ModelX.cs

```diff
@@ -11,13 +11,14 @@ namespace System.ClientModel.Tests.Client.ModelReaderWriterTests.Models
 {
     public class ModelX : BaseModel, IJsonModel<ModelX>
     {
-        public ModelX()
-            : base(null)
+        public ModelX() : base(null)
         {
             Kind = "X";
+            Fields = new List<string>();
+            KeyValuePairs = new Dictionary<string, string>();
         }
 
-        internal ModelX(string kind, string name, int xProperty, int? nullProperty, IList<string> fields, IDictionary<string, string> keyValuePairs, Dictionary<string, BinaryData> rawData)
+        internal ModelX(string kind, string? name, int xProperty, int? nullProperty, IList<string> fields, IDictionary<string, string> keyValuePairs, Dictionary<string, BinaryData> rawData)
```

Comment:
Optional reference type properties become nullable in the serialization constructor.